server/controllers/__test__/project.controller.test.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @jest-environment node
+ */
+import { Response } from 'jest-express';
+
+import { createMock } from '../../models/project';
+import createProject from '../project.controller/createProject';
+
+jest.mock('../../models/project');
+
+describe('project.controller', () => {
+  describe('createProject()', () => {
+    let ProjectMock;
+
+    beforeEach(() => {
+      ProjectMock = createMock();
+    });
+
+    afterEach(() => {
+      ProjectMock.restore();
+    });
+
+    it('fails if create fails', (done) => {
+      const error = new Error('An error');
+
+      ProjectMock
+        .expects('create')
+        .rejects(error);
+
+      const request = { user: {} };
+      const response = new Response();
+
+      const promise = createProject(request, response);
+
+      function expectations() {
+        expect(response.json).toHaveBeenCalledWith({ success: false });
+
+        done();
+      }
+
+      promise.then(expectations, expectations).catch(expectations);
+    });
+
+    it('extracts parameters from request body', (done) => {
+      const request = {
+        user: { _id: 'abc123' },
```

Comment:
should `user` here be within `body`?

Reply:
The [current controller](https://github.com/processing/p5.js-web-editor/blob/df1219b/server/controllers/project.controller.js#L17) extracts the user id from `req.user._id`. I think Passport attaches the user from the logged in session to `req.user`.